src/utils.jl

```diff
@@ -1,5 +1,36 @@
-using Polynomials4ML: natural_indices
+using Polynomials4ML: natural_indices, ScalarPoly4MLBasis, lux
+using LuxCore: AbstractExplicitLayer
 
+struct Radial_basis
+   Rnl::AbstractExplicitLayer
+   Radialspec::Vector{NamedTuple}
+end
+
+Radial_basis(Rnl::AbstractExplicitLayer, spec_Rnl::Union{Vector{Int}, UnitRange{Int64}}) = 
+         Radial_basis(Rnl, [(n = i, ) for i in spec_Rnl])
```

Comment:
Yes, there will be. However, this function simply ensures that if the input specification is a set of integers (which means that the radial basis, in this case, is intended to depend only on n), then the format of the input specification will be changed to a NamedTuple instead of simply a set of integer. if the input specification is indeed n&l-dependent, then it should not be a set of integers so won't call this function.

Reply:
I see - thanks.